### ISSUE TITLE
Add built-in First and Last item functions

### ISSUE THREAD

**Issue by kzu** [dotnet/msbuild#4859]
Sometimes it's useful to take just the first item on a list, or the last. This is currently very cumbersome, to the point that it's "easier" just [write an inline task](http://blog.seravy.com/msbuild-tasks-first-item-in-an-item-group/) to do so. 

A new `First()` and `Last()` item function would simplify things a lot. 

Usage example:

```xml
   <Target Name="Split">
        <ItemGroup>
            <BranchParts Include="release;v1.3.1" />
        </ItemGroup>

        <Message Text="@(BranchParts -> Last())" Importance="high" />
        <Message Text="@(BranchParts -> First())" Importance="high" />
    </Target>
```

Renders:
```
v1.3.0
release
```

**Comment by Rabadash8820** [comment]
I just bumped into this challenge myself, and came across the exact same inline task article 😄 My use case is parsing a version string from a file with the [`ReadLinesFromFile` task](https://docs.microsoft.com/en-us/visualstudio/msbuild/readlinesfromfile-task). The exact format of the file is beyond my control (a Unity3D `ProjectVersion.txt` file), but I know that the first line of the file is in the format `m_EditorVersion: <version>`. `ReadLinesFromFile` returns the file lines as an item collection, so I'd like to be able to call `First` in some way.

**Comment by jrdodds** [comment]
With `First()` (or `Car()`) there should be a `Cdr()` equivalent function added as well.

For a list of `1;2;3`, `First()` would return `1` and `Cdr()` would return `2;3`.

(Car and cdr are from Lisp and cdr is clearly not a good name for this function in MSBuild.)

**Comment by jrdodds** [comment]
Yes. Modeled on System.Linq.Enumerable, there could be `First()`, `Last()`, `Skip(n)`, and `Take(n)`.
